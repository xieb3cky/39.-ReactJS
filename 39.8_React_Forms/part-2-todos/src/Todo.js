import React, { useState } from "react";

function Todo({ id, task = "default", remove }) {

    const handleDelete = () => remove(id);

    return (
        <div>
            <li>{task}</li>
            <button onClick={handleDelete}>X</button>
        </div>
    )
}



export default Todo;