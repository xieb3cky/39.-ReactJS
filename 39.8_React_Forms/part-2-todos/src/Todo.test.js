import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Todo from "./Todo";

it("renders without crashing", function () {
    render(<Todo />);
});

it("matches snapshot", function () {
    const { asFragment } = render(<Todo />);
    expect(asFragment()).toMatchSnapshot();
});


it("runs the delete function on button click", function () {
    const removeMock = jest.fn();
    const { getByText } = render(<Todo remove={removeMock} />);
    const deleteButton = getByText("X");
    fireEvent.click(deleteButton);
    expect(removeMock).toHaveBeenCalled();
});
