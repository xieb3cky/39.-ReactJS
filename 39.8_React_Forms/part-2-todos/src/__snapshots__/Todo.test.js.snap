// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <li>
      default
    </li>
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
